### PR TITLE
remove unnecessary Authorization header on webhook endpoints

### LINKED ISSUE #1898: Remove authorization header on /webhook/callback routes
[download]: https://bintray.com/dv8fromtheworld/maven/JDA/_latestVersion
[guild]: https://discord.gg/0hMr4ce0tIk3pSjp
[stack overflow]: https://stackoverflow.com/questions/tagged/java

## General Troubleshooting

  
- [x] I have checked for similar issues.
- [x] I have updated to the [latest JDA version][download].
- [x] I have checked the branches or the maintainers' PRs for upcoming features.

## Feature Request

Currently JDA sends the Bot Authorization token on the Interactions route /interactions/:ID/callback which isn't needed but if provided still counts against the Bots Global Ratelimit, it would be desirable to not fill up the global ratelimit with requests which do not need to count against the global ratelimit if the header would be explicitly removed for this route 👍 

### Example Use-Case

should be clear but the use case would be to have less issues hitting the Global Ratelimit on bots who heavily use interactions (i had to learn this the hard way).

### PR DESCRIPTION
**Pull request by caneleex**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [x] Internal code
- [ ] Library interface (affecting end-user code) 
- [ ] Documentation
- [ ] Other: \_____ 

Closes #1898 

## Description

this PR removes the Authorization header from callback endpoints requests. I'm not sure which endpoints don't require this, so assistance is welcome

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/internal/requests/Route.java [hunk 1]
```diff
@@ -81,7 +81,7 @@
         public static final Route GET_COMMAND_PERMISSIONS =      new Route(GET, "applications/{application_id}/guilds/{guild_id}/commands/{command_id}/permissions");
         public static final Route EDIT_COMMAND_PERMISSIONS =     new Route(PUT, "applications/{application_id}/guilds/{guild_id}/commands/{command_id}/permissions");
 
-        public static final Route CALLBACK =        new Route(POST,   "interactions/{interaction_id}/{interaction_token}/callback");
+        public static final Route CALLBACK =        new Route(POST,   "interactions/{interaction_id}/{interaction_token}/callback", false);
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/internal/requests/Route.java:84]: This should also be added to all the webhook routes as well as `/gateway`

File: src/main/java/net/dv8tion/jda/internal/requests/Route.java [hunk 2]
```diff
@@ -81,7 +81,7 @@
...
         public static final Route CREATE_FOLLOWUP = new Route(POST,   "webhooks/{application_id}/{interaction_token}");
         public static final Route EDIT_FOLLOWUP =   new Route(PATCH,  "webhooks/{application_id}/{interaction_token}/messages/{message_id}");
         public static final Route DELETE_FOLLOWUP = new Route(DELETE, "webhooks/{application_id}/{interaction_token}/messages/{message_id}");
```
Comment (caneleex) [inline src/main/java/net/dv8tion/jda/internal/requests/Route.java:87]: should all these `webhooks/` endpoints also not require the header?

File: src/main/java/net/dv8tion/jda/internal/requests/Route.java [hunk 3]
```diff
@@ -180,19 +180,19 @@
 
     public static class Webhooks
     {
-        public static final Route GET_WEBHOOK          = new Route(GET,    "webhooks/{webhook_id}");
-        public static final Route GET_TOKEN_WEBHOOK    = new Route(GET,    "webhooks/{webhook_id}/{token}");
-        public static final Route DELETE_WEBHOOK       = new Route(DELETE, "webhooks/{webhook_id}");
-        public static final Route DELETE_TOKEN_WEBHOOK = new Route(DELETE, "webhooks/{webhook_id}/{token}");
-        public static final Route MODIFY_WEBHOOK       = new Route(PATCH,  "webhooks/{webhook_id}");
-        public static final Route MODIFY_TOKEN_WEBHOOK = new Route(PATCH,  "webhooks/{webhook_id}/{token}");
+        public static final Route GET_WEBHOOK          = new Route(GET,    "webhooks/{webhook_id}", false);
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/internal/requests/Route.java:183]: Endpoints without a token in the URL require authorization.

**Review by MinnDevelopment — APPROVED** [review]

**Comment by DV8FromTheWorld** [comment]
I currently don't intend to merge this as it was stated that this is _not_ something that libraries should be handling.
This is a bug in the Discord platform itself.

**Pull request closed without merging**